src/app/users/login.ts

```diff
@@ -64,33 +65,66 @@ export const login = async ({
   await rewardFailedLoginAttemptPerIpLimits(ip)
   await rewardFailedLoginAttemptPerPhoneLimits(phone)
 
-  const user = await UsersRepository().findByPhone(phone)
-  let account: Account
+  let kratosToken: SessionToken
 
-  if (user instanceof CouldNotFindUserFromPhoneError) {
-    subLogger.info({ phone }, "new user signup")
+  const authService = AuthWithPhonePasswordlessService()
+
+  let kratosResult = await authService.login(phone)
+  if (kratosResult instanceof LikelyNoUserWithThisPhoneExistError) {
+    // user has not migrated to kratos or it's a new user
+
+    kratosResult = await authService.createWithSession(phone)
+    if (kratosResult instanceof Error) return kratosResult
     addAttributesToCurrentSpan({ "login.newAccount": true })
-    const userRaw: NewUserInfo = { phone }
-    const account_ = await createAccountForPhoneSchema({
-      newUserInfo: userRaw,
-      config: getDefaultAccountsConfig(),
-    })
-    if (account_ instanceof Error) return account_
-    account = account_
-  } else if (user instanceof Error) {
-    return user
-  } else {
-    subLogger.info({ phone }, "user login")
 
-    const account_ = await AccountsRepository().findByUserId(user.id)
-    if (account_ instanceof Error) return account_
-    account = account_
+    kratosToken = kratosResult.sessionToken
+    const kratosUserId = kratosResult.kratosUserId
+
+    const user = await UsersRepository().findByPhone(phone)
+    if (user instanceof CouldNotFindUserFromPhoneError) {
+      // brand new user
+      subLogger.info({ phone }, "new user signup")
+
+      // TODO: look at where is phone metadata stored
+      const accountRaw: NewAccountInfo = { phone, kratosUserId }
+      const account_ = await createAccountForPhoneSchema({
+        newAccountInfo: accountRaw,
+        config: getDefaultAccountsConfig(),
+      })
+
+      if (account_ instanceof Error) return account_
```

Comment:
what is the suffix `account_` for? Typo? I don't think we have used it anywhere up to now. I have only seen `_account` for variables that aren't going to be used (but it doesn't apply here as it is being returned).